src/ULog.ts

```diff
@@ -169,38 +175,71 @@ export class ULog {
     this.#appendedOffsets = appendedOffsets.map((n) => Number(n)) as [number, number, number];
     const firstOffset = this.#appendedOffsets[0];
 
-    this.#dataEnd = this.#reader.size();
+    this.#dataEnd = reader.size();
     if (firstOffset > 0 && firstOffset < this.#dataEnd) {
       this.#dataEnd = firstOffset;
     }
 
     this.#header = { version, timestamp, flagBits, information, parameters, definitions };
 
-    await this.#createIndex();
+    await this.#createIndex(reader);
   }
 
   async *readMessages(
-    opts: { startTime?: bigint; endTime?: bigint } = {},
+    opts: {
+      startTime?: bigint;
+      endTime?: bigint;
+      includeLogs?: boolean;
+      msgIds?: Set<number>;
+    } = {},
   ): AsyncIterableIterator<DataSectionMessage> {
-    const sortedMessages = this.#timeIndex;
-    if (sortedMessages == undefined) {
+    const includeLogs = opts.includeLogs ?? false;
+    const msgIds = opts.msgIds;
+
+    const reader = new ChunkedReader(this.#fileLike, this.#chunkSize);
+
+    const timeIndex = this.#timeIndex;
+    if (timeIndex == undefined) {
       throw new Error(`Cannot readMessages before createIndex`);
     }
 
-    if (sortedMessages.length === 0) {
+    if (timeIndex.length === 0) {
       return;
     }
 
-    const startTime = opts.startTime ?? sortedMessages[0]![0];
-    const endTime = opts.endTime ?? sortedMessages[sortedMessages.length - 1]![0];
+    const startTime = opts.startTime ?? timeIndex[0]![0];
+    const endTime = opts.endTime ?? timeIndex[timeIndex.length - 1]![0];
 
-    const range = findRange(sortedMessages, startTime, endTime);
+    const range = findRange(timeIndex, startTime, endTime);
     if (range == undefined) {
       return;
     }
 
     for (let i = range[0]; i <= range[1]; i++) {
-      yield sortedMessages[i]![2];
+      const indexEntry = timeIndex[i]!;
```

Comment:
I'd prefer having this more readable without having to remember what index stands for what
```
const [_timestamp, offset, msgId] = timeIndex[i]!;
```